applications/FluidDynamicsApplication/custom_strategies/strategies/residualbased_predictorcorrector_velocity_bossak_scheme_turbulent.h

```diff
@@ -232,6 +272,25 @@ namespace Kratos {
          */
         /*@{ */
 
+        void Initialize(ModelPart& rModelPart) override
+        {
+            KRATOS_TRY;
+
+            BaseType::Initialize(rModelPart);
+
+            const int number_of_elements = rModelPart.NumberOfElements();
+
+#pragma omp parallel for
+            for (int i = 0; i < number_of_elements; i++)
+            {
+                Element& r_element = *(rModelPart.ElementsBegin() + i);
+                r_element.SetValue(DERIVATIVES_EXTENSION,
+                                        Kratos::make_shared<ElementDerivativesExtension>(&r_element));
```

Comment:
one per element?
Would it be possible to use one for all? (I don't think so)

Also why shared and not unique ptr?

Reply:
thanks :) it is possible to use unique ptr :)